### ISSUE TITLE
Way of "getting the tx hash from block to look it up" failing / index-api give random txID since v1.10 upgrade 

### ISSUE THREAD

**Issue by jun0tpyrc** [ava-labs/avalanchego#1433]
A demo tx picked from from explorer
https://avascan.info/blockchain/x/tx/PNwiwiBWEu3k1dZGyExtLcWBx1STkjNScxMqCt1eXvgGeYmiR

=> vertex = Y3pusjr5EbvRafHGCGeyefWGbYKtWxD4HM2XjbMAuR6hETg1P
belongs to #1572 block

```
curl -X POST --data '{
    "jsonrpc": "2.0",
    "method": "avm.getBlockByHeight",
    "params": {
        "height": 1572,
        "encoding": "json"
    },
    "id": 1
}' -H 'content-type:application/json;' 127.0.0.1:9650/ext/bc/X
{"jsonrpc":"2.0","result":{"block":{"parentID":"wyhvwFGUEhESycbPdtxRXDmBr7wpeWrbyiwU9p1xfUji7af3f","height":1572,"time":1682581676,"merkleRoot":"11111111111111111111111111111111LpoYY","txs":[{"unsignedTx":{"networkID":1,"blockchainID":"2oYMBNV4eNHyqk2fjjV5nVQLDbtmNJzq5s3qs3Lo6ftnC6FByM","outputs":[],"inputs":[{"txID":"HshU7FVkAAoNG4b1kiSsSh4fWn19HzmVqWW2iyKecemaoKPGx","outputIndex":1,"assetID":"FvwEAhmxKfeiG8SnEvq42hc6whRyY3EFYAvebMqDNDGCgxN5Z","fxID":"spdxUxVJQbX85MGxMHbKw1sHxMnSqJ3QBzDyDYEP3h6TLuxqQ","input":{"amount":1653529013,"signatureIndices":[0]}}],"memo":"0x","destinationChain":"11111111111111111111111111111111LpoYY","exportedOutputs":[{"assetID":"FvwEAhmxKfeiG8SnEvq42hc6whRyY3EFYAvebMqDNDGCgxN5Z","fxID":"spdxUxVJQbX85MGxMHbKw1sHxMnSqJ3QBzDyDYEP3h6TLuxqQ","output":{"addresses":["X-avax1tnuesf6cqwnjw7fxjyk7lhch0vhf0v95wj5jvy"],"amount":1652529013,"locktime":0,"threshold":1}}]},"credentials":[{"fxID":"spdxUxVJQbX85MGxMHbKw1sHxMnSqJ3QBzDyDYEP3h6TLuxqQ","credential":{"signatures":["0x4cf5a5ae1d2cd75a7284ae42bb60a7af363d6a875ebaa347b8a89bb014d62c151a76ec3a33a31355955ed06febe7751755df88dfb90c53f72a48ac554109cd4200"]}}]}]},"encoding":"json"},"id":1}
```
Problem:
How to get the txId `PNwiwiBWEu3k1dZGyExtLcWBx1STkjNScxMqCt1eXvgGeYmiR` from block
the only txId we can get from block is `HshU7FVkAAoNG4b1kiSsSh4fWn19HzmVqWW2iyKecemaoKPGx` but not anything we can find the tx in explorer/rpc

**Comment by jun0tpyrc** [comment]
indexing api also doesn't have things after cufoff
```
curl --location --request POST 'localhost:9650/ext/index/X/tx' --header 'Content-Type: application/json' --data-raw '{ "jsonrpc": "2.0", "method": "index.getContainerByIndex", "params": { "index":4141251, "encoding": "hex" }, "id": 1 }'
{"jsonrpc":"2.0","error":{"code":-32000,"message":"no container at index 4141251","data":null},"id":1}
```

```
 curl --location --request POST 'localhost:9650/ext/index/X/tx' --header 'Content-Type: application/json' --data-raw '{ "jsonrpc": "2.0", "method": "index.getContainerByIndex", "params": { "index":4141250, "encoding": "hex" }, "id": 1 }'
{"jsonrpc":"2.0","result":{"id":"SWLx2CTYyjYwNUdtAVjTNybHKzYqa3b7i9qDuCa1ke5R2ZhfN","bytes":"0x00000000000000000001ed5f38341e436e5d46e2bb00b45d62ae97d1b050c64bc634ae10626739e35c4b0000000121e67317cbc4be2aeb00677ad6462778a8f52274b9d605df2591b23027a87dff00000007000000001ee00d8000000000000000000000000100000001042615af4434ddbb9f81f1a74e461d90a3efec8c000000012fe32d4c9d368badbd20fefb521ec7ac8ba9dfa27cb01d3508581c7c48f8d9b30000000021e67317cbc4be2aeb00677ad6462778a8f52274b9d605df2591b23027a87dff00000005000000001eef4fc0000000010000000000000008496e7465726e616c00000001000000090000000186f4994bce02663be2e69b15808e1768d5e086e41758e5a050af5a4778c72a6276d86d3ccd038a2978416903a7d6095abd3e96fd6d8bcc14cd86f8144b41de9c00051b167d","timestamp":"2023-04-25T14:59:24.71523653Z","encoding":"hex","index":"4141250"},"id":1}
```

**Comment by jun0tpyrc** [comment]
replacing with new `/ext/index/X/block` => it gives new data, but the `result.id` doesn't look valid txID anymore

```
curl --location --request POST 'localhost:9650/ext/index/X/block' --header 'Content-Type: application/json' --data-raw '{ "jsonrpc": "2.0", "method": "index.getContainerByIndex", "params": { "index":1140, "encoding": "hex" }, "id": 1 }'
{"jsonrpc":"2.0","result":{"id":"2qhDsM8W29rQjvTbRW4ZQFG17isMR4AZutcCUMzaCKgiykRQ8p","bytes":"0x0000000000006a2279d4a321891e423be8b2f4250b24fab143313fd35043c4d8f7dde5c6e0e90000000064498644000000000053808e0000000000000183000000000014b794bbf5fd6192fee2fb084c299c5aa2ed1e51d71b3dadcf40b7cde15a1a5650000000000000047500000000644986440000000000000000000000000000000000000000000000000000000000000000000000010000000000000001ed5f38341e436e5d46e2bb00b45d62ae97d1b050c64bc634ae10626739e35c4b0000000121e67317cbc4be2aeb00677ad6462778a8f52274b9d605df2591b23027a87dff000000070000006a25df4672000000000000000000000001000000018db573053ce1dedfcc1f041ed31a4317ebf308b50000000176dda70c417fa590a2e8461affdefb40a1ccceacf8de900e811d4a12597b0fbb0000000021e67317cbc4be2aeb00677ad6462778a8f52274b9d605df2591b23027a87dff000000050000006a25ee88b200000001000000000000000000000001000000090000000197e205683965bed1c99b201e2bf137fc90e5be2453f6dfb61a363a9c9f98398d0c20ab35137159e7146261a18b1f31921e322cb26973279b6cf02ae0e1003b4e00000000005d06a80a","timestamp":"2023-04-26T20:15:02.505390337Z","encoding":"hex","index":"1140"},"id":1}
```

**Comment by StephenButtolph** [comment]
1. #1408 introduced `txID`s and `blockID`s as part of the json response to `avm.getBlockByHeight`. This will be included in `v1.10.1`.
2. The X-chain is no longer a DAG, so `/ext/index/X/tx` and `/ext/index/X/vtx` are frozen and will no longer index new operations. As discovered here, the last transaction that will ever be indexed `/ext/index/X/tx` will be at index `4141250`.
3. See this [example](https://github.com/ava-labs/avalanchego/blob/535456298046b5c2fbcb95ce36702422b6980c66/indexer/examples/x-chain-blocks/main.go) for how to utilize the `/ext/index/X/block` API (including getting access to X-chain `txID`s.

A useful document to read is also the [X-chain migration doc](https://docs.avax.network/apis/avalanchego/x-chain-migration).

**Comment by MaximIgor** [comment]
Really Thanks
Good friends!!!!
